### ISSUE TITLE
Up axis argument for IfcConvert obj export

### ISSUE THREAD

**Issue by FreakTheMighty** [IfcOpenShell/IfcOpenShell#657]
This feature request was mentioned here https://github.com/IfcOpenShell/IfcOpenShell/issues/421#issuecomment-409810649. I wanted to create a dedicated issue to track this. IfcOpenShell's up axis is `Z`.

The OBJ file format doesn't specify, and has now way to indicate which axis is up. As noted in the comment, software imports OBJ into their internal coordinate system. As such, it would be helpful to export OBJs with an up axis that matches your downstream pipeline. 

The new argument could look like this and be added to the "Serialization options":

`IfcConvert --up-axis Y input.ifc output.obj`

**Comment by aothms** [comment]
Yes it makes sense. Also Collada has an option somewhere in the XML by which it can be specified.

From Z to Y is 90 degree rotation around the X-Axis. So you don't really need to do matrix multiplication. You just need to write X, Z, -Y instead of X, Y, Z (or something similar at least ...)

In the v0.6.0 branch the binary gltf exporter already takes this into account as glb is Y-up. https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/serializers/GltfSerializer.cpp#L168

Would happily accept pull request towards this goal. Not sure if I have time for this soon.

**Comment by Moult** [comment]
For anyone interested in tackling this the order seems to be this based on an obj2ifc script I did:

```
        if self.format == "obj":
            # OBJ swaps Y and Z axis
            return [coordinates[0], -coordinates[2], coordinates[1]]
```

So probably the inverse of that for ifc2obj :)

**Comment by falken10vdl** [comment]
@gisspace. This is related to: IFC file conversion obj file, some cases will be rotated 90 ° #1097 . Did you have apull request for it?
Thanks! It looked you were asking how to submit but then there is no other information if that was done. Can you clarify?

**Comment by aothms** [comment]
There is a setting for OBJ https://github.com/IfcOpenShell/IfcOpenShell/blob/df18108406eaa5fbac2df67e4255fdc9bdf8ac4c/src/ifcgeom/GeometrySerializer.h#L59-L63

**Comment by BIMvoice** [comment]
Closing — this is implemented already: `IfcConvert --y-up` (`src/ifcgeom/GeometrySerializer.h`, wired into `WavefrontObjSerializer.cpp`) swaps the up axis to Y for OBJ output. Verified live: converting the same file with and without the flag produces correctly swapped/signed coordinates (e.g. a vertex's Z and Y components exchange as expected). Not under the exact `--up-axis Y` name originally proposed, but the same functionality. Reopen if you need a different target axis or format.

This comment was written with AI assistance.
